File: src/main/kotlin/org/tendiwa/plane/geometry/straightSkeleton/UnderlyingFace.java
package org.tendiwa.plane.geometry.straightSkeleton;

interface UnderlyingFace {
    Chain startHalfface();

    boolean isHalfface(Chain chain);

    Chain lastAddedChain();

    void increaseNumberOfSkeletonNodes(int d);

    void addNewSortedEnd(Node oneEnd);

    void setLastAddedChain(Chain chain);

    void forgetNodeProjection(Node node);
}

File: src/main/kotlin/org/tendiwa/plane/geometry/straightSkeleton/UnderlyingFace.kt
package org.tendiwa.plane.geometry.straightSkeleton

internal interface UnderlyingFace {
    val startHalfface: Chain

    var lastAddedChain: Chain

    fun isHalfface(chain: Chain): Boolean

    fun increaseNumberOfSkeletonNodes(d: Int)

    fun addNewSortedEnd(end: Node)

    fun forgetNodeProjection(node: Node)
}
